ddtrace/appsec/_asm_request_context.py

```diff
@@ -56,74 +55,125 @@ class ASM_Environment:
     It is contained into a ContextVar.
     """
 
-    def __init__(self, active: bool = False):
-        self.active: bool = active
-        self.span: Optional[Span] = None
-        self.span_asm_context: Optional[contextlib.AbstractContextManager] = None
+    def __init__(self, span: Optional[Span] = None):
+        self.root = not in_context()
+        if self.root:
+            core.add_suppress_exception(BlockingException)
+        if span is None:
+            self.span: Span = core.get_item("call")
```

Comment:
```suggestion
            self.span: Span = core.get_item(core.get_item("call_key"))
```
This will ensure that this code grabs the current context's span regardless of the name it's stored under.

Reply:
Maybe we could add a get_span in core?